lib/matplotlib/tests/test_rcparams.py

```diff
@@ -422,3 +422,64 @@ def test_rcparams_reset_after_fail():
                 pass
 
         assert mpl.rcParams['text.usetex'] is False
+
+
+def test_if_rctemplate_is_up_to_date():
+    # This tests if the matplotlibrc.template file
+    # contains all valid rcParams.
+    dep1 = mpl._all_deprecated
+    dep2 = mpl._deprecated_set
+    deprecated = list(dep1.union(dep2))
+    #print(deprecated)
+    path_to_rc = mpl.matplotlib_fname()
+    with open(path_to_rc, "r") as f:
+        rclines = f.readlines()
+    missing = {}
+    for k,v in mpl.defaultParams.items():
+        if k[0] == "_":
+            continue
+        if k in deprecated:
+            continue
+        if "verbose" in k:
+            continue
+        found = False
```

Comment:
This can be written more compact:
~~~
if not any(k in line for line in lines):
    missing[k] = v
~~~
Note, that `k in line` is a bit imprecise because it would also match values or comment.

Reply:
Correct. I think any better solution here would eventually lead to writing one single test and match the parsed file agains the defaultParams. Depends on if people think this should be done or not.

Reply:
I'm a big fan of "incrementally better", so I don't want to enforce a full parsing test. Your partial test is already a big improvement. Maybe you could add a short comment to clarify that this test is not expected to be 100%. And of course, use the compact form :smiley:.